### PR TITLE
Sg delete not matching cidr for port

### PR DESCRIPTION
**Pull request by harelsa**
for a chosen port, the bot deletes all rules that  do not match the chosen cider\

created for DFR-823

### CODE REVIEW

File: bots/Bots.md [hunk 1]
```diff
@@ -383,6 +388,29 @@ What it does: Deletes all ingress and egress rules from a SG
...
 
+
+## sg\_delete\_not\_matching\_cidr
+
+What it does: Deletes all rules on a security group with a certain port when the cidr is not matching input
+
+Usage: sg_delete_not_matching_cidr port scope direction
+
+Parameters:
+    port: integer
```
Comment (RavidAtias) [inline bots/Bots.md:399]: change integer to number

File: bots/sg_delete_not_matching_cidr.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    ec2_resource = boto_session.resource('ec2')
+    # get sg by sg_id of the entity
+    sg = ec2_resource.SecurityGroup(sg_id)
+    port = str(port)
+
+    # going through all the user's sg and check for mismatch
+    for rule in entity[f'{direction}Rules']:
+        if rule[PORT_FROM] <= int(port) <= rule[PORT_TO]:
+            # port found , if the scope doesn't match , rule will be deleted
+            if rule[SCOPE] != scope:
```
Comment (RavidAtias) [inline bots/sg_delete_not_matching_cidr.py:106]: I think find scope in range is bit more complex.. you can ask Guy he knows more than i

File: bots/Bots.md (hunk 1 above)
Comment (harelsa) [reply bots/Bots.md:399]: done

File: bots/sg_delete_not_matching_cidr.py (hunk 2 above)
Comment (harelsa) [reply bots/sg_delete_not_matching_cidr.py]: fixed after consulting with Guy , it is more complex

File: bots/sg_delete_not_matching_cidr.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    ec2_resource = boto_session.resource('ec2')
+    # get sg by sg_id of the entity
+    sg = ec2_resource.SecurityGroup(sg_id)
+    port = str(port)
+
+    # going through all the user's sg and check for mismatch
+    for rule in entity[f'{direction}Rules']:
+        if rule[PORT_FROM] <= int(port) <= rule[PORT_TO]:
+            # port found , if the scope doesn't match , rule will be deleted
+            in_scope = utils.is_scope_contained_by_other_ipv4(rule[SCOPE], scope)
```
Comment (RavidAtias) [inline bots/sg_delete_not_matching_cidr.py:61]: No need to keep in variable

Comment (RavidAtias) [inline bots/sg_delete_not_matching_cidr.py:61]: No need to keep in variable

File: bots_utils.py [hunk 4]
```diff
@@ -39,3 +40,98 @@ def verify_scope_is_cidr(rule):
...
+Check if cider is completely inside scope(other cidr), if it does returns true  
+"""
+
+
+def is_scope_contained_by_other_ipv4(scope, other):
+    n1 = ipaddress.IPv4Network(scope)
+    n2 = ipaddress.IPv4Network(other)
+
+    inside = n2.supernet_of(n1)
+    if inside:
```
Comment (RavidAtias) [inline bots_utils.py:70]: You can use ternary operator:
 return True if inside else False

File: bots_utils.py [hunk 5]
```diff
@@ -39,3 +40,98 @@ def verify_scope_is_cidr(rule):
...
+"""
+Check if cider is completely inside scope, if it does returns true 
+"""
+
+
+def is_scope_contained_by_other_ipv6(scope, other):
+    n1 = ipaddress.IPv6Network(scope)
+    n2 = ipaddress.IPv6Network(other)
+    inside = n2.supernet_of(n1)
+    if inside:
```
Comment (RavidAtias) [inline bots_utils.py:85]: You can use ternary operator:
return True if inside else False

**Review by RavidAtias — CHANGES_REQUESTED** [review]

File: bots_utils.py (hunk 5 above)
Comment (harelsa) [reply bots_utils.py:85]: done

File: bots_utils.py (hunk 4 above)
Comment (harelsa) [reply bots_utils.py:70]: done

File: bots/sg_delete_not_matching_cidr.py (hunk 3 above)
Comment (harelsa) [reply bots/sg_delete_not_matching_cidr.py:61]: done

File: bots_utils.py (hunk 5 above)
Comment (RavidAtias) [reply bots_utils.py:85]: I don't see changes, did you push?

**Review by RavidAtias — APPROVED** [review]
Please add to the UI

**Pull request merged**